internal/server/models/module/version.go

```diff
@@ -8,12 +8,12 @@ import (
 
 type Version struct {
 	entity.Entity
-	ModuleID     uuid.UUID
+	ModuleID     uuid.UUID `gorm:"size:256"`
 	Module       Module
 	Version      string       `gorm:"not null"`
 	Location     string       `gorm:"not null"`
-	Providers    []Provider   `gorm:"foreignKey:ParentID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
-	Dependencies []Dependency `gorm:"foreignKey:ParentID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
+	Providers    []Provider   `gorm:"foreignKey:ParentID;constraint:OnUpdate:NO ACTION,OnDelete:NO ACTION"`
+	Dependencies []Dependency `gorm:"foreignKey:ParentID;constraint:OnUpdate:NO ACTION,OnDelete:NO ACTION"`
```

Comment:
I don't know about those two. Currently, those fields are unused since I plan to use them for the documentation feature. If the cascade on deletion is removed, we risk collecting them as garbage in the database as modules are deleted over time.
Can you explain a bit about what errors you see that made you think removing cascade here will fix it?

Reply:
Thanks @valentindeaconu. Here is what I am seeing with CASCADE

gorm generated migration:
```sql
CREATE TABLE "module_providers" ("id" nvarchar(256),"created_at" datetimeoffset,"updated_at" datetimeoffset,"parent_id" nvarchar(256),"name" nvarchar(MAX) NOT NULL,"namespace" nvarchar(MAX) NOT NULL,"source" nvarchar(MAX) NOT NULL,"version" nvarchar(MAX) NOT NULL,PRIMARY KEY ("id"),CONSTRAINT "fk_module_versions_providers" FOREIGN KEY ("parent_id") REFERENCES "module_versions"("id") ON DELETE CASCADE ON UPDATE CASCADE,CONSTRAINT "fk_module_submodules_providers" FOREIGN KEY ("parent_id") REFERENCES "module_submodules"("id") ON DELETE CASCADE ON UPDATE CASCADE)
```
Failed to execute query. Error: Introducing FOREIGN KEY constraint 'fk_module_submodules_providers' on table 'module_providers' may cause cycles or multiple cascade paths. Specify ON DELETE NO ACTION or ON UPDATE NO ACTION, or modify other FOREIGN KEY constraints.
Could not create constraint or index. See previous errors.

Reply:
Oh great, so it complains about both being set to cascade. I don't really care about `ON UPDATE CASCADE`, but `ON DELETE CASCADE` should remain as it handles the clean-up if a module version gets deleted. Can you try setting the FK constraint to the following?
```
constraint:OnUpdate:NO ACTION,OnDelete:CASCADE"
```

Reply:
I did give it a go with 
```
constraint:OnUpdate:NO ACTION,OnDelete:CASCADE
```
However still no joy. I think the error sql server gives is a bit misleading. If any form of potential cascade cycle it refuses to create the constraint. Potentially would have to rethink the schema slightly or implement another way to cleanup orphaned records to be able to support this. Tricky one..

![cascades](https://github.com/terralist/terralist/assets/13116496/a37b85c2-9077-497e-88dd-b6fc409003b8)

Reply:
Okay, then don't bother with it. As I said, they're not used at this time, so by the time we add support for the documentation we should figure out a way to clean them up.

Thanks for looking it up!